docs: add release notes for 0.20.0 (#6113)
determined-ai/determined: docs/release-notes.rst

 Release Notes
###############

**************
 Version 0.20
**************

Version 0.20.0
==============

**Release Date:** February 28, 2023

**Breaking Changes**

> -  Cluster: ``resources.agent_label`` task option and agent config ``label`` option are no longer

Reviewer: ```suggestion
-  Cluster: The ``resources.agent_label`` task option and ``label`` agent config option are no longer
```

>    supported and will be ignored. If you are not explicitly using these options, or only use single

Reviewer: ```suggestion
   supported and will be ignored. If you are not explicitly using these options, or only use a single
```

   empty or non-empty label value per resource pool, no changes are necessary. Otherwise, cluster

>    admins should create a resource pool for each existing ``resource_pool`` + ``agent_label``

Reviewer: ```suggestion
   admins should create a resource pool for each existing ``resource_pool``/``agent_label``
```

>    combination, and reconfigure agents to use these new pools. Cluster users should update their

Reviewer: ```suggestion
   combination and reconfigure agents to use these new pools. Cluster users should update their
```

   tasks to use the new resource pool names.

**Bug Fixes**

-  Model Registry: Fix an issue where a model with versions from multiple workspaces could have its
   versions modified by a user with edit access to only a single one of those workspaces.
-  WebUI: Patch an issue where logging out would not properly redirect to the login page.

> -  Fix a bug where in certain cases could cause the cluster's job queue page to crash.

Reviewer: ```suggestion
-  WebUI: Fix a bug where the cluster's job queue page could crash in certain cases.
```

**Improvements**

> -  Agents: The master configuration ``agent_reattach_enabled`` is always enabled and agents now will
>    always reattach containers on restarts.

Reviewer: ```suggestion
-  Agents: The master configuration ``agent_reattach_enabled`` is always enabled and agents will now
   always reattach containers on restart.
```

-  Kubernetes: The cluster information page now takes resource quotas into account if there are any
   on relevant namespaces.

-  RBAC: Model registry models and commands that are inaccessible to the user will appear as
   uneditable. Previously, users could attempt the action and would encounter a permission denied
   error.

> -  CLI: When listing tensorboards, show ``workspaceName`` instead of ``workspaceId`` for better

Reviewer: ```suggestion
-  CLI: When listing TensorBoards, show ``workspaceName`` instead of ``workspaceId`` for better
```

   readability and prevent N/A values from appearing.

**New Features**

> -  RBAC: Following on the initial RBAC support added in 0.19.7 the enterprise edition of Determined

Reviewer: ```suggestion
-  RBAC: Following on the initial RBAC support added in 0.19.7, the enterprise edition of Determined
```

   (`HPE Machine Learning Development Environment
   <https://www.hpe.com/us/en/solutions/artificial-intelligence/machine-learning-development-environment.html>`_)

>    has added support for Role-Based Access Control over new entities:

Reviewer: ```suggestion
   has added support for role-based access control (RBAC) over new entities:
```

>    -  JupyterLab Notebooks, Tensorboards, Shells, and Commands are now housed under workspaces.

Reviewer: ```suggestion
   -  Notebooks, TensorBoards, shells, and commands are now housed under workspaces.
```

      Access to these tasks can now be restricted by role.

>    -  Model Registry: models are now associated with workspaces. Models can be moved between

Reviewer: ```suggestion
   -  Model Registry: Models are now associated with workspaces. Models can be moved between
```
